dali/pipeline/executor/executor.h

```diff
@@ -373,6 +374,16 @@ class DLL_PUBLIC Executor : public ExecutorBase, public QueuePolicy {
     }
   }
 
+  /**
+   * @brief Traverses the Graph to check if there is a _conditional.split with _if_stmt argument
+   * specified, indicating that the `if` statement was used and the graph is operating in
+   * conditional mode (enable_conditionals=True).
+   *
+   * TODO(klecki): Consider adding a specific callback to Pipeline so the frontend can indicate
+   * it directly.
+   */
+  void DetectConditionals();
```

Comment:
Maybe it would be good to add some super small test for it just to cover it and have it running in the tests for future reference.

Reply:
Sounds like a splendid idea.